### PR TITLE
Changed valid sessions criteria and retrieving only required sessions.

### PR DESCRIPTION
**Pull request by AlbertoCasasOrtiz**
Fixes https://github.com/stanfordnmbl/opencap-viewer/issues/55.

I have changed the criteria of what a valid session is according to https://github.com/stanfordnmbl/opencap-viewer/issues/55.

Also, I detected we were querying all sessions from database, sorting them and then filtering; instead of directly querying only sessions complying with the filter, and then sorting. This PR also fixes it.

### CODE REVIEW

File: mcserver/views.py [hunk 1]
```diff
@@ -175,12 +175,19 @@ def valid(self, request):
...
             quantity = request.data['quantity']
-        if(quantity == -1):
-            # Note the use of `get_queryset()` instead of `self.queryset`
-            sessions = self.get_queryset().order_by("-created_at").annotate(trial_count=Count('trial')).filter(trial_count__gte=3, user=request.user)
-        else:
-            # Note the use of `get_queryset()` instead of `self.queryset`
-            sessions = self.get_queryset().order_by("-created_at").annotate(trial_count=Count('trial')).filter(trial_count__gte=3, user=request.user)[:request.data['quantity']]
+
+        # Note the use of `get_queryset()` instead of `self.queryset`
+        sessions = self.get_queryset().order_by("-created_at").annotate(trial_count=Count('trial')).filter(trial_count__gte=1, user=request.user)
```
Comment (suhlrich) [inline mcserver/views.py:180]: the `order_by("-created_at")` is redundant with `get_queryset()`

**Review by suhlrich — COMMENTED** [review]
Looks good. Can we merge into dev to test it first?

**Comment by antoinefalisse** [comment]
@AlbertoCasasOrtiz looks like there is a conflict? Might taking a look. We can test when merged in dev. Thanks.

**Comment by AlbertoCasasOrtiz** [comment]
@antoinefalisse The conflict should be solved now. I am going to merge.

**Comment by AlbertoCasasOrtiz** [comment]
On my side seems to work, but I only have 1 session. Can you check if it is working with more than one session, and check if the Load all button works (hopefully faster than before). @antoinefalisse @suhlrich

**Comment by antoinefalisse** [comment]
Thanks, I will test. I guess we need to wait for the "delete" feature to be ready before merging into main.

**Pull request merged**